### PR TITLE
Add signale

### LINKED ISSUE #26: Update logs with signale logger
Update all logs with appropriate [signale logger](https://github.com/klauscfhq/signale) logs. Also add timer logs where the time taken is logged.

### PR DESCRIPTION
**Pull request by kapillamba4**
Closes #26

### CODE REVIEW

File: app/actions/fetchData.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
+import signale from 'signale';
 import fetch from 'node-fetch';
 import { userPayload } from './payload';
 import fetchReposList from './fetchReposList';
 import fetchRepo from './fetchRepo';
 import fetchPullRequests from './fetchPullRequests';
 
 const fetchData = async username => {
-  console.log(`Fetching data from github: ${username}`);
+  signale.pending(`Fetching data from github: ${username}`);
```
Comment (0xnmn) [inline app/actions/fetchData.js]: remove this log

File: app/actions/fetchData.js [hunk 2]
```diff
@@ -27,7 +30,8 @@ const fetchData = async username => {
     const profile = ((await profileResponse.json()).data || {}).user; // user profile
     if (!profile) {
       // return 404 if user not found on github
-      console.log(`User non found on github: ${username}`);
+      signale.fatal(`User non found on github: ${username}`);
```
Comment (0xnmn) [inline app/actions/fetchData.js]: use warning logger

File: app/actions/fetchData.js [hunk 3]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
     data.repos = reposList.map((repoBasicData, index) => ({ ...repoBasicData, ...reposDetailedDataCollection[index] }));
+    signale.success(`Fetched ${data.repos.length} repos detailed data for user: ${username}`);
 
-    console.log(
-      `Fetched ${data.repos.length} repos detailed data for user: ${username} in ${new Date() - startReposDataTime}ms`,
-    );
+    signale.timeEnd('Fetch Repositories Data');
 
-    // set current time
-    data.time = new Date();
```
Comment (0xnmn) [inline app/actions/fetchData.js]: THIS BREAKS EVERYTHING! Don't delete this.

File: app/actions/fetchData.js [hunk 4]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
+    signale.timeEnd('Fetch Github Data');
```
Comment (0xnmn) [inline app/actions/fetchData.js]: also add a complete log. `Request for fresh data for ${username}`

File: app/actions/fetchData.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
 import fetchRepo from './fetchRepo';
 import fetchPullRequests from './fetchPullRequests';
 
 const fetchData = async username => {
-  console.log(`Fetching data from github: ${username}`);
+  signale.pending(`Fetching data from github: ${username}`);
   let data = {}; // final data to return
 
   try {
+    signale.time('Fetch Github Data');
```
Comment (0xnmn) [inline app/actions/fetchData.js]: `Fetching data from github: ${username}`

File: app/actions/fetchData.js [hunk 6]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
     /*
       Accumulate list of all pull requests
     */
-    const startPRTime = new Date();
+    signale.time('Fetch Pull Requests');
```
Comment (0xnmn) [inline app/actions/fetchData.js]: `fetching pull requests for ${username}`

File: app/actions/fetchData.js [hunk 7]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
+    signale.timeEnd('Fetch Pull Requests');
     data.pullRequests = pullRequests;
 
     /*
       Accumulate list of all repos with basic details including
       in series and return array of repos (acm).
       Fetch basic details for 100 repos at a time.
     */
-    const startReposListTime = new Date();
+    signale.time('Fetch Repo List');
```
Comment (0xnmn) [inline app/actions/fetchData.js]: `fetching repos list for ${username}`

File: app/actions/fetchData.js [hunk 8]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
+    signale.success(`Fetched ${reposList.length} repos list for user: ${username}`);
+    signale.timeEnd('Fetch Repo List');
 
     /*
       For each repo in the repo list fetch
       detailed stats data in parallel.
     */
-    const startReposDataTime = new Date();
+
+    signale.time('Fetch Repositories Data');
```
Comment (0xnmn) [inline app/actions/fetchData.js]: `fetching repo data for ${username}`

File: app/actions/fetchPullRequests.js [hunk 9]
```diff
@@ -6,8 +7,8 @@ const fetchPullRequests = async (username, acm = [], endCursor = null) => {
     Accumulate all pull requests in series and return
     array of pull requests (acm). Fetch 100 PR at a time
   */
-  const startTime = new Date();
 
+  signale.time('Fetch Pull requests');
```
Comment (0xnmn) [inline app/actions/fetchPullRequests.js]: remove the timer here.

File: app/actions/fetchPullRequests.js [hunk 10]
```diff
@@ -28,7 +29,8 @@ const fetchPullRequests = async (username, acm = [], endCursor = null) => {
   const cursor = prData.data.user.pullRequests.pageInfo.endCursor;
   // current set of pull requests
   const pullRequests = prData.data.user.pullRequests.nodes || [];
-  console.log(`Successfully ${pullRequests.length} pull requests ${cursor} fetched in ${new Date() - startTime}ms`);
+  signale.success(`Successfully ${pullRequests.length} pull requests ${cursor} fetched`);
```
Comment (0xnmn) [inline app/actions/fetchPullRequests.js]: `fetched ${pullRequests.length} pull requests for ${username}`

File: app/actions/fetchRepo.js [hunk 11]
```diff
@@ -1,11 +1,13 @@
+import signale from 'signale';
 import fetch from 'node-fetch';
 import { repoPayload } from './payload';
 
 const fetchRepo = async (owner, name, username, uid) => {
   /*
     Fetch detailed repo data
   */
-  const startTime = new Date();
+  signale.time('Fetch Details for a Repository');
```
Comment (0xnmn) [inline app/actions/fetchRepo.js]: remove the timer here

File: app/actions/fetchRepo.js [hunk 12]
```diff
@@ -28,7 +30,9 @@ const fetchRepo = async (owner, name, username, uid) => {
   }
 
   const repo = repoData.data.repository || {};
-  console.log(`Successfully repo ${username}@${owner}/${name} detailed data fetched in ${new Date() - startTime}ms`);
+  signale.success(`Successfully repo ${username}@${owner}/${name} detailed data fetched`);
```
Comment (0xnmn) [inline app/actions/fetchRepo.js]: `fetched repo ${username}@${owner}/${name} data`

File: app/actions/fetchRepos.js [hunk 13]
```diff
@@ -1,11 +1,13 @@
...
+import signale from 'signale';
 import { reposPayload } from './payload';
 
 const fetchRepos = async (username, uid, endCursor = null) => {
   /*
     Fetch next 20 repos based on cursor
   */
-  const startTime = new Date();
+
+  signale.time('Fetch Repositories (max 20)');
```
Comment (0xnmn) [inline app/actions/fetchRepos.js]: remove the timer here.

File: app/actions/fetchRepos.js [hunk 14]
```diff
@@ -29,7 +31,8 @@ const fetchRepos = async (username, uid, endCursor = null) => {
 
   // collection of 20 repos
   const repos = reposData.data.user.repositories.nodes;
-  console.log(`Successfully ${repos.length} repos collection ${endCursor} fetched in ${new Date() - startTime}ms`);
+  signale.success(`Successfully ${repos.length} repos collection ${endCursor} fetched`);
```
Comment (0xnmn) [inline app/actions/fetchRepos.js]: `fetched ${repos.length} repos collection for ${username}`

File: app/actions/fetchReposList.js [hunk 15]
```diff
@@ -7,8 +8,8 @@ const fetchReposList = async (username, acm = [], endCursor = null) => {
     in series and return array of repos (acm).
     Fetch basic details for 100 repos at a time.
   */
-  const startTime = new Date();
 
+  signale.time('Fetch Repositories List (max 100)');
```
Comment (0xnmn) [inline app/actions/fetchReposList.js]: remove the timer here.

File: app/actions/fetchReposList.js [hunk 16]
```diff
@@ -29,7 +30,8 @@ const fetchReposList = async (username, acm = [], endCursor = null) => {
   const cursor = reposList.data.user.repositories.pageInfo.endCursor;
   // current set of repos
   const repos = reposList.data.user.repositories.nodes;
-  console.log(`Successfully ${repos.length} repos list fetched in ${new Date() - startTime}ms`);
+  signale.success(`Successfully ${repos.length} repos list fetched`);
```
Comment (0xnmn) [inline app/actions/fetchReposList.js]: `fetched ${repos.length} repos list for ${username}`

File: app/resolvers/query/user.js [hunk 17]
```diff
@@ -21,9 +22,9 @@ export default async (parent, { username, fresh }, { db }, info) => {
     */
 
     if (!findUser) {
-      console.log(`Data for user ${username} not present.`);
+      signale.error(`Data for user ${username} not present.`);
```
Comment (0xnmn) [inline app/resolvers/query/user.js]: use note logger

File: app/resolvers/query/user.js [hunk 18]
```diff
@@ -21,9 +22,9 @@ export default async (parent, { username, fresh }, { db }, info) => {
...
     } else {
-      console.log(`Fresh data requested for ${username}.`);
+      signale.success(`Fresh data requested for ${username}.`);
```
Comment (0xnmn) [inline app/resolvers/query/user.js]: use star logger

File: app/actions/fetchData.js [hunk 19]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
     // detailed repo stats
     const reposDetailedDataCollection = await Promise.all(
       reposList.map(({ owner, name }) => fetchRepo(owner, name, username, data.uid)),
     );
 
     // accumulate repos data and add to user data
     data.repos = reposList.map((repoBasicData, index) => ({ ...repoBasicData, ...reposDetailedDataCollection[index] }));
+    signale.success(`Fetched ${data.repos.length} repos detailed data for user: ${username}`);
```
Comment (0xnmn) [inline app/actions/fetchData.js]: remove this log

File: app/actions/fetchData.js [hunk 20]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
...
     /*
       Accumulate list of all repos with basic details including
       in series and return array of repos (acm).
       Fetch basic details for 100 repos at a time.
     */
-    const startReposListTime = new Date();
+    signale.time('Fetch Repo List');
     const reposList = await fetchReposList(username);
-    console.log(`Fetched ${reposList.length} repos list for user: ${username} in ${new Date() - startReposListTime}ms`);
+    signale.success(`Fetched ${reposList.length} repos list for user: ${username}`);
```
Comment (0xnmn) [inline app/actions/fetchData.js]: remove the log

File: app/actions/fetchData.js [hunk 21]
```diff
@@ -55,41 +59,42 @@ const fetchData = async username => {
     /*
       Accumulate list of all pull requests
     */
-    const startPRTime = new Date();
+    signale.time('Fetch Pull Requests');
     const pullRequests = await fetchPullRequests(username);
-    console.log(`Fetched ${pullRequests.length} pull requests for user: ${username} in ${new Date() - startPRTime}ms`);
+    signale.success(`Fetched ${pullRequests.length} pull requests for user: ${username}`);
```
Comment (0xnmn) [inline app/actions/fetchData.js]: remove the log

**Review by 0xnmn — CHANGES_REQUESTED** [review]

**Comment by kapillamba4** [comment]
@hereisnaman 
image

**Comment by 0xnmn** [comment]
@kapillamba4 great work, you just forget removing the 'Fetch Pull requests' timer.

**Comment by 0xnmn** [comment]
@championswimmer merging this pr from @kapillamba4 for 50 bounty points.

**Pull request merged**